docs/root/faq/rolling_upgrades.rst

```diff
@@ -0,0 +1,124 @@
+.. _common_configuration_rolling_upgrades:
+
+How do I handle rolling upgrades and transient failures?
```

Comment:
@ramaraochavali FWIW this FAQ seems much less about rolling upgrades and much more about transient failures? A nit, but perhaps remove the reference to rolling upgrades in the question and rename the RST to something about transient failures? (Very great writeup though!!)

Reply:
Actually my point was to highlight that it works for both transient and rolling upgrades. I think this statement covers that "resilient to realities such as rolling upgrades, dynamic infrastructure, and network failures" -So probably we can remove it from question and rename if that makes it more clearer.
Will change.